Use stubparser dependency correctly.

wmdietl opened the pull request:
Instead of #1467 we should depend on the stubparser built from source.
https://github.com/typetools/stubparser/pull/7 sets the stubparser project properly.

wmdietl commented:
Two missing pieces:
- integrate with release scripts
- go through manual and whenever annotation-tools is checked-out&built, also build stubparser

wmdietl commented:
This PR will fail until https://github.com/typetools/stubparser/pull/7 is merged.

mernst commented:
@wmdietl , you said:

> Two missing pieces:
> 
>  * integrate with release scripts
>  * go through manual and whenever annotation-tools is checked-out&built, also build stubparser

Are these things you plan to do, or are they things you want someone else to do?

wmdietl commented:
As we don't make a separate release of the stubparser, I think the release integration will be easy. I would appreciate if @smillst could look at that part.

I will look through the manual on Wednesday and see whether any more changes are needed.

mernst commented:
When running Javadoc, it needs a pointer to some location with the Stub Parser (JavaParser) classes.

wmdietl commented:
https://github.com/typetools/checker-framework/pull/1476/commits/b075678e7b48037ad763eb6f2fa12204bd5bb374 adds the stubparser to the building from source instructions.

wmdietl commented:
https://github.com/typetools/checker-framework/pull/1476/commits/c1e08d8fda9e5500a18f105df2dad2f275378032 adds the stubparser.lib on more classpaths, in particular for javadoc.

wmdietl commented:
@smillst Please let me know if you want me to look into the release scripts. Everything else is addressed.

smillst commented:
@wmdietl If you have time to look at them today that would be helpful.  I can test the changes tonight.

wmdietl commented:
@smillst  I added https://github.com/typetools/checker-framework/pull/1476/commits/778b7f947ef4e69e9c0f88f1099c282edbe6e5bb 
StubParser should require integration similar to plume-lib. I think I added it in all those places.
Eventually, we should discuss whether StubParser needs its own release.

smillst commented on release/release_build.py:
```diff
@@ -66,6 +67,7 @@ def clone_or_update_repos(auto):
 
     clone_from_scratch_or_update(LIVE_PLUME_LIB, PLUME_LIB, clone_from_scratch, False)
     clone_from_scratch_or_update(LIVE_PLUME_BIB, PLUME_BIB, clone_from_scratch, False)
+    clone_from_scratch_or_update(LIVE_STUBPARSER, STUBPARSER, clone_from_scratch, False)
```
This clones the stub parser, but doesn't build it.  (Plume lib is built when it is cloned in the post-merge hook.)

smillst replied:
I'm working on building it.

smillst replied:
Added in 28c101e.

wmdietl commented:
Can we merge this? My comment about the release process can also be addressed the next time we will need to change the version number.

smillst commented:
I'm fine with merging.  We should add a script to stubparser that does the copying so that the instructions in the manual, release scripts, and travis script don't all have to be updated separately.

wmdietl commented:
I would prefer if all projects just have `.travis-build-without-test.sh` as build instructions and if that is used from the release scripts. But let's discuss that as a separate PR.
